src/InitialConditions/baro_wave.jl

```diff
@@ -129,10 +129,30 @@ function center_initial_condition_baroclinic_wave(
     end
     # TODO: Include ability to handle nonzero initial cloud condensate
 
+    precip_kwargs =
+        if precip_model isa NoPrecipitation &&
+           !(turbconv_model isa TC.EDMFModel)
+            NamedTuple()
+        else
+            (; q_rai = FT(0), q_sno = FT(0))
```

Comment:
Should our tracers for rain be the same as other tracers in the dycore. I.e. be rho * q_something ?

Reply:
Probably. This is just a peel off from #916, but it's introducing a pretty major performance penalty due to inference failure

Reply:
:/ Does that mean we don't want to merge it? Do you have any ideas how to help the compiler there?

Reply:
Not as is, it's slowing some jobs down _significantly_. I think the best course of action is to somehow make this optional, and merge it while only adding a single job that exercises it (maybe the perf job?). This way, it doesn't impact everyone, but we can keep an eye on / experiment with it.